generated/nifake/tests/test_session.py

```diff
@@ -78,14 +82,55 @@ def test_init_with_error(self):
             assert e.code == test_error_code
             assert e.description == test_error_desc
 
-    # TODO(marcoskirsch): This should test that when close errors it raises.
-    # def test_close_errors(self):
+    def test_close_with_error(self):
+        test_error_code = -1
+        test_error_desc = 'Test'
+        self.patched_library.niFake_close.side_effect = self.side_effects_helper.niFake_close
+        session = nifake.Session('dev1')
+        self.side_effects_helper['close']['return'] = test_error_code
+        self.patched_library.niFake_GetError.side_effect = self.side_effects_helper.niFake_GetError
+        self.side_effects_helper['GetError']['errorCode'] = test_error_code
+        self.side_effects_helper['GetError']['description'] = test_error_desc
+        try:
+            session.close()
+            assert False
+        except nifake.Error as e:
+            assert e.code == test_error_code
+            assert e.description == test_error_desc
+        self.patched_library.niFake_close.assert_called_once_with(SESSION_NUM_FOR_TEST)
 
-    # TODO(marcoskirsch): This should test that when init errors it raises.
-    # def test_session_context_manager_error_on_init
+    def test_session_context_manager_init_with_error(self):
+        test_error_code = -1
+        test_error_desc = 'Test'
+        self.patched_library.niFake_InitWithOptions.side_effect = self.side_effects_helper.niFake_InitWithOptions
+        self.side_effects_helper['InitWithOptions']['return'] = test_error_code
+        self.side_effects_helper['InitWithOptions']['vi'] = SESSION_NUM_FOR_TEST
+        self.patched_library.niFake_GetError.side_effect = self.side_effects_helper.niFake_GetError
+        self.side_effects_helper['GetError']['errorCode'] = test_error_code
+        self.side_effects_helper['GetError']['description'] = test_error_desc
+        try:
+            with nifake.Session('dev1') as session:
+                assert type(session) == nifake.Session
```

Comment:
Change this to `assert False` since you never expect to get here. You can remove the `assert False` that is outside the with statement

Reply:
I agree, but given that I am using a context manager it creates a variable called session. If I do not call something with session variable I get a flake 8 failure.  Unless you have other suggestions,  I cannot remove this function.

Reply:
Okay, then just move the assert False into the with block.

Reply:
Fixed